packages/noir-ethereum-api/test/e2e.test.ts

```diff
@@ -1,51 +1,85 @@
 import { beforeAll, describe, expect, it } from 'vitest';
 import { incHexStr } from '../src/util/string.js';
-import { verifyStorageProof, readProof, readWitnessMap, readInputMap, circuit } from '../src/main.js';
-import { updateNestedField } from '../src/util/object.js';
-import { InputMap, WitnessMap, abiEncode } from '@noir-lang/noirc_abi';
+import { circuit, readInputMap, readProof, readWitnessMap } from '../src/main.js';
+import { copy, updateNestedField } from '../src/util/object.js';
+import { abiEncode, InputMap, WitnessMap } from '@noir-lang/noirc_abi';
+
+import { Account, privateKeyToAccount } from 'viem/accounts';
+import ultraVerifier from '../../../contracts/out/UltraVerifier.sol/UltraVerifier.json';
+import { Address, Hex } from 'viem';
+import { assert } from '../src/util/assert.js';
+import { verifyStorageProofInSolidity } from '../src/ethereum/verifier.js';
+import { AnvilClient, createAnvilClient } from './ethereum/anvilClient.js';
 
 const PROOF_PATH = '../../proofs/main.proof';
 const INPUT_MAP_PATH = '../../circuits/main/Verifier.toml';
+const ANVIL_TEST_ACCOUNT_PRIVATE_KEY = '0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80';
+const VERIFICATION_GAS_LIMIT = 475455n;
```

Comment:
```suggestion
const VERIFICATION_GAS_LIMIT = 475_455n;
```

If this is an exact value - let's round it up to 500k. Every proof has different cal data and because 0s and 1s have different cost in call data - this introduces a lot of randomness